### PR TITLE
Rotate the project nginx log by logrotate

### PR DESCRIPTION
**Pull request by mark-juwai**
## Summary of changes

- _Code/resources that have been added, updated or removed. Please provide rationale for changes whenever applicable_

## How to Test

- `$ cd tests`
- `$ vagrant up`
- `$ vagrant ssh`
- `$ less /etc/logrotate.d/test`
- `$ sudo logrotate /etc/logrotate.d/test -d -f`

@tjoelsson Who else can I assign the PR to? There are few assignees for this repo.

### CODE REVIEW

File: meta/main.yml [hunk 1]
```diff
@@ -12,4 +12,5 @@ galaxy_info:
   #   versions:
   #   - 2013.09
   galaxy_tags: [amazon, centos6, nginx, php, laravel]
-dependencies: []
+dependencies:
+  - role: retr0h.logrotate
```
Comment (tjoelsson) [inline meta/main.yml]: You can make this a conditional dependency

File: tasks/main.yml [hunk 2]
```diff
@@ -17,6 +17,23 @@
...
     mode: 0775
 
+- name: Rotation for project nginx log file
+  logrotate:
+    name: '{{ application_name }}'
+    path: '/var/log/nginx/{{ application_name }}/*.log'
+  args:
+    options:
+      - 'daily'
+      - 'rotate 14'
```
Comment (tjoelsson) [inline tasks/main.yml]: CNC said they wanted weekly retention

File: defaults/main.yml [hunk 3]
```diff
@@ -1,3 +1,5 @@
 ---
 deploy_user: '{{ ansible_ssh_user }}'
 deploy_group: '{{ ansible_ssh_user }}'
+
+laravel_log_rotate: true
```
Comment (tjoelsson) [inline defaults/main.yml]: Should specify it means nginx log, not application log. How about `laravel_rotate_nginx_log`?

**Review by tjoelsson — CHANGES_REQUESTED** [review]

File: meta/main.yml (hunk 1 above)
Comment (mark-juwai) [reply meta/main.yml]:
```
dependencies:
  - { role: retr0h.logrotate, when: laravel_rotate_nginx_log }
```
Is this what you want? If yes, how we set the variable when downloading the role?

File: tasks/main.yml (hunk 2 above)
Comment (tjoelsson) [inline tasks/main.yml]: Sorry, I just checked the web servers. We have 30 days for juwai.com. Should probably keep the same. Maybe it's best to create a variable for this.

File: meta/main.yml (hunk 1 above)
Comment (tjoelsson) [reply meta/main.yml]: I think download will always happen

File: tasks/main.yml [hunk 4]
```diff
@@ -17,6 +17,23 @@
...
     mode: 0775
 
+- name: Rotation for project nginx log file
+  logrotate:
+    name: '{{ application_name }}'
+    path: '/var/log/nginx/{{ application_name }}/*.log'
+  args:
+    options:
+      - 'daily'
+      - 'rotate 7'
```
Comment (tjoelsson) [inline tasks/main.yml]: Let's create a variable for this, with 30 days as default

**Comment by mark-juwai** [comment]
@tjoelsson Comments addressed, please check again. Thank you.

**Review by tjoelsson — APPROVED** [review]

**Comment by mark-juwai** [comment]
@upbeta01 Please review when you get time. Thanks.

File: tasks/main.yml [hunk 5]
```diff
@@ -17,6 +17,23 @@
     group: 'nginx'
     mode: 0775
 
+- name: Rotation for project nginx log file
+  logrotate:
+    name: '{{ application_name }}'
+    path: '/var/log/nginx/{{ application_name }}/*.log'
```
Comment (upbeta01) [inline tasks/main.yml:23]: Not sure if it's a good idea to have a wildcard on the line. Though, we wanted to apply log rotate to any log file.
Would it be cool to have it on a `list` (ie [ access.log, error.log ])?

More readable and for maintainability.

**Comment by upbeta01** [comment]
Rest of the codes -- are good to me :+1:

**Comment by upbeta01** [comment]
NOTE: The `logrotate` role dependency is using a minimum version of 2.1 and the base role https://github.com/juwai/ansible-role-laravel is written on a 1.9 version. Hopefully, no built-in variable is used on this `role` that's depreciated on 2.1 so things run smoothly.

Lastly, we might want to pin down the version of this external `role` so servers new and old can have a unified installation.

File: tasks/main.yml (hunk 5 above)
Comment (mark-juwai) [reply tasks/main.yml:23]: I don't think it's a bad idea. Nginx logrotate config also uses wildcard, so we don't have to update playbook if there is any other log file comes.

**Comment by mark-juwai** [comment]
There is no ansible version problem for now. Sure we can consistent it later if necessary.

The thumb up above means you approved the PR? 😄

**Comment by upbeta01** [comment]
@mark-juwai - Can we now start pinning versions for the external `role`? That was my last point on the NOTE section.

**Comment by mark-juwai** [comment]
@upbeta01 OK, please advise how to do that.

**Comment by mark-juwai** [comment]
I noticed most of our roles were written in `1.9.*`, I suggest to have a dedicated ticket to fix that. What do you think? @tjoelsson

File: meta/main.yml [hunk 6]
```diff
@@ -12,4 +12,5 @@ galaxy_info:
   #   versions:
   #   - 2013.09
   galaxy_tags: [amazon, centos6, nginx, php, laravel]
-dependencies: []
+dependencies:
+  - { role: retr0h.logrotate, when: laravel_rotate_nginx_log }
```
Comment (upbeta01) [inline meta/main.yml:16]: `{ role: retr0h.logrotate,2.0, when: laravel_rotate_nginx_log }`

- should pin it to the `role` 2.0 version.

Comment (mark-juwai) [reply meta/main.yml:16]: Their 2.0 version is not working well, we do need 2.0.1.

**Comment by tjoelsson** [comment]
Are you guys referring to `min_ansible_version: 1.9` or what it says in the readme? We can update the min_ansible_version for sure. It should be 2.1.

File: meta/main.yml (hunk 6 above)
Comment (upbeta01) [reply meta/main.yml:16]: Ok, if 2.0.1 is what we need. Then we should go with it.

**Comment by mark-juwai** [comment]
@upbeta01 `min_ansible_version` updated.

**Comment by upbeta01** [comment]
Reviewed, all good. :+1:  (Approved)

**Comment by mark-juwai** [comment]
```
ERROR! the role 'retr0h.logrotate,2.0.1' was not found in /Users/juwai/Projects/mobile-api-v2-ansible/roles:/Users/juwai/Projects/mobile-api-v2-ansible:/etc/ansible/roles:/Users/juwai/Projects/mobile-api-v2-ansible/roles
```
Still have error while running the role.

**Comment by upbeta01** [comment]
Let's skip pinning of version for now. Let's figure it out when we have time.

**Comment by mark-juwai** [comment]
OK, thank you. @upbeta01 
I reverted my last commit. Still good? @upbeta01 @tjoelsson

**Review by tjoelsson — APPROVED** [review]

**Pull request merged**